c8/tenant.py

```diff
@@ -192,9 +191,13 @@ def create_tenant(self, email, passwd='', dclist=[], extra={}):
         data = {'name': name}
         data['email'] = email
         data['passwd'] = passwd
-        data['extra'] = extra
         if dclist != '':
             data['dcList'] = dclist
+        if plan:
+            data['plan'] = plan
+            data['contact'] = contact
+            data['metadata'] = metadata
+            data['attribution'] = attribution
```

Comment:
if `plan` is selected there needs to be validation for the other three arguments: contact, metadata, attribution.

`contact` should be a dictionary with specific keys in it, required as minimum from the API.